src/main/java/com/easypost/easyvcr/interactionconverters/HttpUrlConnectionInteractionConverter.java

```diff
@@ -0,0 +1,117 @@
+package com.easypost.easyvcr.interactionconverters;
+
+import com.easypost.easyvcr.Censors;
+import com.easypost.easyvcr.requestelements.HttpInteraction;
+import com.easypost.easyvcr.requestelements.Request;
+import com.easypost.easyvcr.requestelements.Response;
+import com.easypost.easyvcr.requestelements.Status;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+
+import static com.easypost.easyvcr.internalutilities.Tools.readFromInputStream;
+
+/**
+ * The interaction converter to convert Http(s)UrlConnection requests/responses to/from EasyVCR requests/responses.
+ */
+public final class HttpUrlConnectionInteractionConverter extends BaseInteractionConverter {
+    /**
+     * Convert a HttpURLConnection request to an EasyVCR request.
+     * @param connection The HttpURLConnection request.
+     * @param censors The censors to apply to the request.
+     * @return The EasyVCR request.
+     */
+    public Request createRecordedRequest(HttpURLConnection connection, Censors censors) {
+        try {
+            // collect elements from the connection
+            String uriString = connection.getURL().toString();
+            Map<String, List<String>> headers = connection.getRequestProperties();
+            String method = connection.getRequestMethod();
+
+            // apply censors
+            uriString = censors.applyQueryParametersCensors(uriString);
+            headers = censors.applyHeadersCensors(headers);
+
+            // create the request
+            Request request = new Request();
+            request.setMethod(method);
+            request.setUri(new URI(uriString));
+            request.setHeaders(headers);
+            // TODO: Capture and censor body
+            // request.setBody(body);
```

Comment:
Remove the comment and uncomment tine 47

Reply:
This damn line opened a whole can of worms.

It's been dealt with.